### ISSUE TITLE
Wanderlust can't parse From filed

### ISSUE THREAD

**Issue by catap** [wanderlust/wanderlust#237]
Greeting,

When email contains a form field like:
```
From: Riccardo Mottola via macports-dev <macports-dev@lists.macports.org>
```

it can't parse it, and I have empy line instead.

**Comment by ikazuhiro** [comment]
Please describe how you confirmed the header field is not parsable.  On Summary buffer, Message buffer, other buffer or procedure?

**Comment by catap** [comment]
@ikazuhiro on summary buffer I haven't see anything inside from field. When I press `A` to reply to this email, from was also missed.

**Comment by catap** [comment]
@ikazuhiro here how summary looks like for that email:

![image](https://github.com/wanderlust/wanderlust/assets/37775/c3f906c8-9a45-480b-b483-3ca60a56d8d9)

and when I reply to it:

![image](https://github.com/wanderlust/wanderlust/assets/37775/b9586bee-bc9c-473c-92d2-a09d84da22f0)

this email was send to public mail list, so I can share it without any issue. Here it is: https://gist.github.com/catap/d1eae0e6615ed63576cb36b22e8fd30b

**Comment by ikazuhiro** [comment]
Not reproducible for me.  It seems that you fail to build msgdb.  Do you modify or add advice to msgdb related functions?  If so, restore to original functions and rebuild msgdb.

**Comment by catap** [comment]
Nope, I haven't touch msgdb related functions at all. The only hacks which left and which I need to sort is relate to templates.

Here my config: https://gist.github.com/catap/24db737593ee44b687a9aee60337bc22 from which I've cleaned up list of mail-list and accounts.

It runs against current master without any additional hacks. But it based on doomemacs changes, if I need to put exact version here it is: https://github.com/doomemacs/doomemacs/blob/37d2cffb60cf3fde4b28545426ca70a47edde3ee/modules/email/wanderlust/config.el

**Comment by ikazuhiro** [comment]
1. When you reply to the message, draft's header fields shows expected value?
2. What is your `wl-summary-from-function`'s value?
3. Evaluate below code and check your message entity's value

```elisp
(let ((folder-name "+draft") ;; Set your folder name
      (number 123)           ;;  Set your message number
      charset msgdb entity)
  (setq folder (elmo-make-folder folder-name t)
	entity (elmo-message-entity folder number)
	charset (modb-entity-handler-mime-charset
		 (elmo-message-entity-handler entity)))
  (list
   entity
   (elmo-message-entity-field entity 'from)
   (funcall wl-summary-from-function (elmo-message-entity-field entity 'from))
   ))
```

**Comment by catap** [comment]
1. Yes, Message buffer has correct `From` field, and draft has correct `To` field as well.
2.  `wl-summary-default-from`
3. It fails as:
```
Debugger entered--Lisp error: (scan-error "Unbalanced parentheses" 955 1)
  scan-sexps(956 -1)
  forward-sexp(-1)
  elisp--preceding-sexp()
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

**Comment by ikazuhiro** [comment]
Your Emacs seems to return `""` by `(wl-summary-default-from "Riccardo Mottola via macports-dev <macports-dev@lists.macports.org>")`.  It is unexpected.

1. What is your `wl-use-petname` and `wl-summary-get-petname-function` value?
2. Evaluate following code and show result.

```elisp
(funcall wl-summary-get-petname-function
	 "Riccardo Mottola via macports-dev <macports-dev@lists.macports.org>")
```

**Comment by catap** [comment]
Indeed, I do have `""` inside pet name for this list. Is it possible to use value from field instead of petname?

**Comment by catap** [comment]
`(setq wl-use-petname nil)` fixed issue, thanks.

**Comment by ikazuhiro** [comment]
You can remove entry by `@` on Summary buffer.  Or, do you have any purpose for that you have such empty entry?

**Comment by catap** [comment]
@ikazuhiro I write to this mail list time to time, and keep it in address book to easy find it.

A user in quesrtion has domain with DMARC / SPF policy which prevents it delivery from mail-list server, so, it replaces From to the mail list address and name like "Real User via Mail List".

So, I expect to see in Summary "Real User via Mail List", but also to have such record in address book to easy find it.

**Comment by ikazuhiro** [comment]
Try to override `wl-address-get-petname-1` with following one.
```elisp
(defun wl-address-get-petname-1 (string)
  (let ((address (downcase (wl-address-header-extract-address string)))
	petname)
    (when (and (setq petname (gethash address wl-address-petname-hash))
	       (> (length petname) 0))
      petname)))
```

**Comment by catap** [comment]
@ikazuhiro thanks, that works!

**Comment by catap** [comment]
Thanks!
